File: 0_liste&dict/C++/lc38.cpp
// Count and Say 

class Solution {
public:
    string countAndSay(int n){
        string result = "1";

        for (int j = 1; j < n; ++j) {
            string current = "";
            int count = 1;
            for (int e = 1; e < result.size(); ++e) {
                if (result[e] == result[e - 1]) {
                    ++count;
                } else {
                    current += to_string(count) + result[e - 1];
                    count = 1;
                }
            }
            current += to_string(count) + result.back();
            result = current;
        }

        return result;
    }
};

File: 0_strings/lc38.py
# Count and Say (Medium)

class Solution:
    def countAndSay(self, n: int) -> str:
        result = "1"
        for j in range(n-1):
            current = ""
            count = 1
            for e in range(1,len(result)):
                if result[e] == result[e - 1] :
                    count += 1
                else:
                    current += str(count) + result[e - 1]
                    count = 1
            current += str(count) + result[-1]
            result = current
        return result
